Automattic/jetpack
# Plans: use the Jetpack Plans package in the Jetpack plugin, and deprecate existing class

Fixes #31213

## Proposed changes:

In this PR, we start using the package in the Jetpack plugin, and deprecate the existing class. No features should be impacted by the change, everything should work just like before.

### Other information:


## Jetpack product discussion

* pdWQjU-4i-p2
* p9dueE-7pi-p2
* 1202726932197503-as-1202726932197515

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

Lots of things to test here. I would also recommend tailing the site's logs as you test, to ensure no calls of `Jetpack_Plan` (now returning deprecation notices) remains.

* Start by connecting your site to WordPress.com.
* Check all the features where files are touched.
